### PR TITLE
Rename classes

### LINKED ISSUE #39: Rename classes
11 9 17 16 21 office lens

### PR DESCRIPTION
**Pull request by bushw011**
It makes a simple call to the database that updates the classname. The UI for this is an edit button next to the class title that only appears for teachers. This button allows them to do an inline (no dialog popup) edit of the name. Closes #39

|localhost_9000_class_p7w93wrki44pzqxnkiqw iphone 6|
localhost_9000_class_p7w93wrki44pzqxnkiqw iphone 6 1|
localhost_9000_class_p7w93wrki44pzqxnkiqw iphone 6 2|
|---|---|---|

### CODE REVIEW

File: src/app/class-component/class.component.ts [hunk 1]
```diff
@@ -96,6 +98,19 @@ export class ClassComponent implements OnInit, OnDestroy {
         console.log("class destroyed");
     }
 
+    public updateClass(): void {
```
Comment (floogulinc) [inline src/app/class-component/class.component.ts]: should probably be called renameClass

File: src/app/class/class.service.ts [hunk 2]
```diff
@@ -114,6 +114,12 @@ export class ClassService {
             });
     }
 
+    public editClass(classId: string, newClassName: string){
```
Comment (floogulinc) [inline src/app/class/class.service.ts]: should probably be called renameClass

**Review by EthanUphoff — DISMISSED** [review]
Looks good to me, runs well, no bugs found.

**Review by floogulinc — CHANGES_REQUESTED** [review]
The menu icon shouldn't move when editing a card on mobile as it does in the screenshot.

File: src/app/class-component/class.component.css [hunk 3]
```diff
@@ -13,3 +13,7 @@
 #class-save-button{
     width: 2.5%;
 }
+.mat-form-field.class-edit-field{
```
Comment (floogulinc) [inline src/app/class-component/class.component.css]: This only needs to select the class, not `.mat-form-field`

**Review by floogulinc — CHANGES_REQUESTED** [review]

File: src/app/class-component/class.component.html [hunk 4]
```diff
@@ -4,8 +4,8 @@
         <mat-card>
             <div layout="row" layout-align="space-between center">
                 <mat-card-title flex id="decks-title" *ngIf="this.currentClass">
-                    <mat-form-field id="class-edit-field" *ngIf="editing">
-                        <input matInput *ngIf="this.canEdit" #input type="text" placeholder="Class Name" [(ngModel)]="this.currentClass.name">
+                    <mat-form-field class="class-edit-field" id="class-edit-field" *ngIf="editing">
```
Comment (floogulinc) [inline src/app/class-component/class.component.html]: We don't need both an ID and a class

**Review by floogulinc — CHANGES_REQUESTED** [review]

**Review by floogulinc — DISMISSED** [review]

**Comment by floogulinc** [comment]
We should merge this in first and use the new component in renaming decks (#20 and #89)

**Review by floogulinc — APPROVED** [review]
Looks good now!

**Pull request merged**